src/test/unit/EigenPodManagerUnit.t.sol

```diff
@@ -117,220 +85,438 @@ contract EigenPodManagerUnitTests is Test, EigenPodPausingConstants {
+    }
+
+    /*******************************************************************************
+                                     Setters
+    *******************************************************************************/
+
+    function test_setMaxPods() public {
+        // Set max pods
+        uint256 newMaxPods = 0;
+        cheats.expectEmit(true, true, true, true);
+        emit MaxPodsUpdated(eigenPodManager.maxPods(), newMaxPods);
+        cheats.prank(unpauser);
+        eigenPodManager.setMaxPods(newMaxPods);
+
+        // Check storage update
+        assertEq(eigenPodManager.maxPods(), newMaxPods);
+    }
+
+    function testFuzz_updateBeaconChainOracle_revert_notOwner(address notOwner) public {
+        cheats.assume(notOwner != initialOwner);
+        cheats.prank(notOwner);
+        cheats.expectRevert("Ownable: caller is not the owner");
+        eigenPodManager.updateBeaconChainOracle(IBeaconChainOracle(address(1)));
+    }
+
+    function test_updateBeaconChainOracle() public {
+        // Set new beacon chain oracle
+        IBeaconChainOracle newBeaconChainOracle = IBeaconChainOracle(address(1));
+        cheats.prank(initialOwner);
+        cheats.expectEmit(true, true, true, true);
+        emit BeaconOracleUpdated(address(newBeaconChainOracle));
+        eigenPodManager.updateBeaconChainOracle(newBeaconChainOracle);
+
+        // Check storage update
+        assertEq(address(eigenPodManager.beaconChainOracle()), address(newBeaconChainOracle));
+    }
+}
+
+contract EigenPodManagerUnitTests_CreationTests is EigenPodManagerUnitTests, IEigenPodManagerEvents {
+
+    function test_createPod() public {
+        // Get expected pod address and pods before
+        IEigenPod expectedPod = eigenPodManager.getPod(defaultStaker);
+        uint256 numPodsBefore = eigenPodManager.numPods();
+
+        // Create pod
+        cheats.expectEmit(true, true, true, true);
+        emit PodDeployed(address(expectedPod), defaultStaker);
+        eigenPodManager.createPod();
+
+        // Check pod deployed
+        _checkPodDeployed(defaultStaker, numPodsBefore);
+    }
+
+    function test_createPod_revert_alreadyCreated() public deployPodForStaker(defaultStaker) {
+        cheats.expectRevert("EigenPodManager.createPod: Sender already has a pod");
+        eigenPodManager.createPod();
+    }
+
+    function test_createPod_revert_maxPodsCreated() public {
+        // Write numPods into storage. Num pods is at slot 153
+        bytes32 slot = bytes32(uint256(153)); 
+        bytes32 value = bytes32(eigenPodManager.maxPods());
+        cheats.store(address(eigenPodManager), slot, value);
+
+        //Create pod & expect revert based on maxPods size
+        if (eigenPodManager.maxPods() == type(uint256).max) {
+            // Arithmetic over/underflow not working with foundry
+            cheats.expectRevert();
+        } else {
+            cheats.expectRevert("EigenPodManager._deployPod: pod limit reached");
+        }
```

Comment:
since we _are_ setting maxPods to the max uint256 in the setUp, this won't ever actually hit the `else` block / check that this reversion is happening.
I think it probably makes sense to reduce `maxPods` first, or otherwise make sure we are checking both paths here?